pkg/apis/pipeline/v1beta1/param_conversion.go

```diff
@@ -61,13 +69,22 @@ func (p *Param) convertFrom(ctx context.Context, source v1.Param) {
 }
 
 func (v ParamValue) convertTo(ctx context.Context, sink *v1.ParamValue) {
-	sink.Type = v1.ParamType(v.Type)
+	if v.Type != "" {
```

Comment:
Thanks to the discussion with @Yongxuanzhang I changed both the conversion for `paramValue` and `paramSpec`. But I was wondering in which case would we bump into the conversion of an empty string input? Thought the webhook should already taken care of that in the earlier stage?

Reply:
This can happen with resources already stored in etcd without a type set (that were created before we added defaulting logic for param values)